backend/src/main/java/gov/cdc/usds/simplereport/config/ApiUserAwareGraphQlContextBuilder.java

```diff
@@ -95,14 +104,30 @@ private Subject subjectFromCurrentUser() {
    */
   private DataLoaderRegistry buildDataLoaderRegistry() {
     DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
-    DataLoader<UUID, PatientPreferences> patientPreferencesLoader =
-        new DataLoader<>(
+
+    dataLoaderRegistry.register(
+        PatientPreferences.DATA_LOADER,
+        new DataLoader<UUID, PatientPreferences>(
             personIds ->
                 supplyAsync(
                     () ->
                         patientPreferencesRepository.findAllAndCoalesceEmptyByPersonInternalIdIn(
-                            personIds)));
-    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesLoader);
+                            personIds))));
+
+    dataLoaderRegistry.register(
+        PatientDataResolver.LAST_TEST_DATA_LOADER,
+        new DataLoader<UUID, TestEvent>(
+            patientIds ->
+                supplyAsync(
+                    () -> {
+                      Map<UUID, TestEvent> found =
+                          testEventRepository.findLastTestsByPatient(patientIds).stream()
+                              .collect(Collectors.toMap(TestEvent::getInternalId, s -> s));
```

Comment:
I don't love that we're still making multiple queries to get the last test information (though `1+1` queries is definitely preferable to `n+1`!). Is there any way to make hibernate do a join for us on the initial patients query?

Reply:
Hm... I think so, yes – we'd have to move it in the GraphQL schema from person > lastTest > dateAdded, directly onto person, and call it `lastTestDate`. Then we'd add `lastTestDate` to the PatientDataResolver, and I could use a BatchLoader to put the queries together

Reply:
☝️  I've tried this approach; it did not magically combine the queries, and stayed `1+1`. So I'm leaving this as-is, and leaving `dateAdded` as requested below to minimize churn

Reply:
Thanks for trying! graphql-java pushes the DataLoader pattern pretty hard, so leaving this as-is is Officially The Right Thing To Do™️.